lib/turbolinks.rb

```diff
@@ -27,6 +27,20 @@ def referer
           end
           alias referrer referer
         end
+
+        ActionDispatch::Routing::Redirect.class_eval do
+          def call_with_turbolinks(env)
+            status, headers, body = call_without_turbolinks(env)
+            session = ActionDispatch::Request::Session.find(env)
```

Comment:
I don't like this, but Rails 4.0 doesn't let us access the request object it creates:
https://github.com/rails/rails/blob/4-0-stable/actionpack/lib/action_dispatch/routing/redirection.rb#L18
Unlike 4.2:
https://github.com/rails/rails/blob/4-2-stable/actionpack/lib/action_dispatch/routing/redirection.rb#L24